packages/bits/demo/src/components/demo/common/unit-conversion-service/unit-conversion-service-separate-unit-display/unit-conversion-service-separate-unit-display.example.component.ts

```diff
@@ -20,7 +20,11 @@ export class UnitConversionServiceSeparateUnitDisplayExampleComponent implements
     public onNumberChange(num: number): void {
         this.num = num;
         const conversion: IUnitConversionResult = this.unitConversionService.convert(this.num, UnitBase.Standard, 2);
-        this.valueDisplay = this.unitConversionService.getValueDisplay(conversion);
         this.unitDisplay = this.unitConversionService.getUnitDisplay(conversion, "hertz");
+        if(this.unitDisplay) {
+            this.valueDisplay = this.unitConversionService.getValueDisplay(conversion);
+        } else {
+            this.valueDisplay = this.unitConversionService.getScientificDisplay(conversion);
```

Comment:
I think we need to update the unit as well:

```suggestion
            this.unitDisplay = unitConversionConstants[unit][0];
            this.valueDisplay = this.unitConversionService.getScientificDisplay(conversion);
```